src/strategy/indicators/cmf.ts

```diff
@@ -0,0 +1,30 @@
+// Chaikin Money Flow
+export class CMF {
+  public static calculate(periods: Record<string, number>[], length: number) {
+    if (periods.length >= length) {
+      let moneyFlowVolume = 0
+      let sumOfVolume = 0
+    
+      periods.slice(0,length).forEach((curr) => {
```

Comment:
you could already unpack the curr within the forEach, if you do the following, which is more easy to read

>       periods.slice(0,length).forEach(({close, low, high, volume}) => {
>        moneyFlowVolume += (volume * (close - low - (high - close))) / (high - low)
 >       sumOfVolume += volume
>      })

Reply:
Or you could 1-line it.

```typescript
const [moneyFlowVolume, sumOfVolume] = periods.slice(0, length).reduce((([a, b], { close: c, low: l, high: h, volume: v }) => (a += (v * (c - l - (h - c))) / (h - l)), (b += v), [a, b]), [0, 0])
```

Reply:
thats maybe a one liner but for me somehow hard to understand. i would prefer my code :-D

Reply:
:trollface: